presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -422,6 +422,10 @@ protected static CacheQuota generateCacheQuota(HiveSplit hiveSplit)
             }
         }
 
+        if (!hiveColumns.isEmpty() && hiveColumns.stream().allMatch(hiveColumnHandle -> hiveColumnHandle.getColumnType() == AGGREGATED)) {
```

Comment:
if all column handles are AGGREGATED, we should not enter HivePageSourceProvider, right?
How about use checkState, with error message like:
`Invalid processing state. Partial aggregation should be processed in AggregatedParquetPageSource or AggregatedOrcPageSource`

Reply:
Right - this error can happen when the metadata defines the table as orc/parquet, but there is at least one file that is in text/rc format. This is potentially a user-facing error message and I wouldn't expect a user to know about AggregatedParquetPageSource or AggregatedOrcPageSource. IMO it makes more sense to point out that the file format here is not supported and suggest one way to work around this. Let me know if you disagree

Reply:
get it. how about some error message like:
`table $table input format is: $inputformat, but this file is: $filetype`